scripts/module_list/tests/data/lmod_mock.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
```

Comment:
add `set -u` the produce an error when the expected environment variable is not defined?

Reply:
Some env variables are intentionally not defined, for example:

![image](https://github.com/hpcugent/vsc_user_docs/assets/95340152/c5a94a5a-a5cd-495a-99d0-a7ae72d74505)

If `$mod_args` is empty, it will just be an empty string, due to the leading and trailing ".